xdr/transaction_meta.go

```diff
@@ -16,3 +20,43 @@ func (transactionMeta *TransactionMeta) OperationsMeta() []OperationMeta {
 		panic("Unsupported TransactionMeta version")
 	}
 }
+
+// GetDiagnosticEvents returns all contract events emitted by a given operation.
+func (t *TransactionMeta) GetDiagnosticEvents() ([]DiagnosticEvent, error) {
+	switch t.V {
+	case 1:
+		return nil, nil
+	case 2:
+		return nil, nil
+	case 3:
+		var diagnosticEvents []DiagnosticEvent
+		var contractEvents []ContractEvent
+		if sorobanMeta := t.MustV3().SorobanMeta; sorobanMeta != nil {
+			diagnosticEvents = sorobanMeta.DiagnosticEvents
+			if len(diagnosticEvents) > 0 {
+				// all contract events and diag events for a single operation(by it's index in the tx) were available
```

Comment:
nit: since we're here, might as well fixup the typo
```suggestion
				// all contract events and diag events for a single operation (by its index in the tx) were available
```